scripts/gulp/frontend-shared.js

```diff
@@ -25,22 +27,29 @@ const buildFrontendSharedJs = () => {
   );
 };
 
-const linkFrontendShared = done => {
-  // Setup a symlink from the client to the frontend-shared package.
-  cp.spawn('yarn', ['link'], {
-    env: process.env,
-    cwd: './frontend-shared',
-  }).on('exit', () => {
-    cp.spawn('yarn', ['link', '@hypothesis/frontend-shared'], {
-      env: process.env,
-      stdio: 'inherit',
-    }).on('exit', () => {
-      done();
-    });
-  });
+const buildFrontendSharedTypes = async () => {
+  // nb. If the options get significantly more complex, they should be moved to
+  // a `tsconfig.json` file.
+  await run('node_modules/.bin/tsc', [
+    '--allowJs',
+    '--declaration',
+    '--emitDeclarationOnly',
+    '--outDir',
+    'frontend-shared/lib',
+    'frontend-shared/src/index.js',
+  ]);
+};
+
+const linkFrontendShared = async () => {
+  // Make @hypothesis/frontend-shared available for linking in other projects.
+  await run('yarn', ['link'], { cwd: './frontend-shared' });
```

Comment:
A side effect of running `yarn link` like this is that some warnings that were previously hidden are now visible. One warning is about a missing license, which is addressed in https://github.com/hypothesis/client/pull/2954. The other is about running `yarn link` being redundant if the package is already linked, which I suggest we deal with separately.